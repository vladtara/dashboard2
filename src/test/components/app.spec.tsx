import { render } from "@testing-library/react";
import App, { IAppProps } from "../../components/app";

const props: IAppProps = {
  name: "App Test",
  icon: "bug_report",
  url: "#",
  displayURL: "test",
};

describe("app.tsx", () => {
  it("Tests app rendering with newTab=true", () => {
    const { baseElement } = render(
      <App
        name={props.name}
        icon={props.icon}
        url={props.url}
        displayURL={props.displayURL}
        newTab={true}
      />,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("Tests app rendering with newTab=false", () => {
    const { baseElement } = render(
      <App
        name={props.name}
        icon={props.icon}
        url={props.url}
        displayURL={props.displayURL}
        newTab={false}
      />,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("Tests app rendering without newTab", () => {
    const { baseElement } = render(
      <App
        name={props.name}
        icon={props.icon}
        url={props.url}
        displayURL={props.displayURL}
      />,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
