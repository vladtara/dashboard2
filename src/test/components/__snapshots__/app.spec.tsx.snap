// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app.tsx Tests app rendering with newTab=false 1`] = `
<body>
  <div>
    <a
      class="sc-dkzDqf gHJogt"
      href="#"
    >
      <div
        class="sc-hKMtZM kjGwCV"
      >
        <i
          class="sc-gsnTZi ctUcHQ"
        >
          bug_report
        </i>
      </div>
      <div
        class="sc-eCYdqJ bcsQqv"
      >
        <div
          class="sc-jSMfEi bkhPVp"
        >
          App Test
        </div>
        <p
          class="sc-gKXOVf EDUWD"
        >
          test
        </p>
      </div>
    </a>
  </div>
</body>
`;

exports[`app.tsx Tests app rendering with newTab=true 1`] = `
<body>
  <div>
    <a
      class="sc-dkzDqf gHJogt"
      href="#"
      rel="noopener noreferrer"
      target="_blank"
    >
      <div
        class="sc-hKMtZM kjGwCV"
      >
        <i
          class="sc-gsnTZi ctUcHQ"
        >
          bug_report
        </i>
      </div>
      <div
        class="sc-eCYdqJ bcsQqv"
      >
        <div
          class="sc-jSMfEi bkhPVp"
        >
          App Test
        </div>
        <p
          class="sc-gKXOVf EDUWD"
        >
          test
        </p>
      </div>
    </a>
  </div>
</body>
`;

exports[`app.tsx Tests app rendering without newTab 1`] = `
<body>
  <div>
    <a
      class="sc-dkzDqf gHJogt"
      href="#"
    >
      <div
        class="sc-hKMtZM kjGwCV"
      >
        <i
          class="sc-gsnTZi ctUcHQ"
        >
          bug_report
        </i>
      </div>
      <div
        class="sc-eCYdqJ bcsQqv"
      >
        <div
          class="sc-jSMfEi bkhPVp"
        >
          App Test
        </div>
        <p
          class="sc-gKXOVf EDUWD"
        >
          test
        </p>
      </div>
    </a>
  </div>
</body>
`;
